app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,45 @@
+class MarketsController < ApplicationController
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    @market = Market.new(market_edit_params[:market])
+    if @market.save
+      redirect_to markets_path
+    else
+      render :new
+    end
+  end
+
+  def index
+    @markets = Market.all
+    @market = Market.first
+    @vendor = Vendor.first
+  end
+
```

Comment:
I can't figure out what @market and @vendor are used for in the view. Maybe I'm just missing it- double check that you actually need this.